pkg/flink/resources.go

```diff
@@ -156,28 +157,47 @@ func (fc *FlinkCluster) updateJobSpec(taskCtx FlinkTaskContext, taskManagerRepli
 		AfterJobCancelled: flinkOp.CleanupActionDeleteCluster,
 	}
 
-	if strings.HasPrefix(out.JarFile, "gs://") {
+	urls := make([]string, len(artifacts))
+	useGcs := true
+	for i, s := range artifacts {
+		urls[i] = s.String()
+		if useGcs && s.Scheme != "gs" {
+			useGcs = false
+		}
+	}
+
+	// XXX(julient) I don't like that this would just silently fail if the condition is not satisfied
+	if useGcs && len(artifacts) > 0 {
 		//TODO(regadas): add job resources to the config
 		resourceList := corev1.ResourceList{
 			corev1.ResourceCPU:    resource.MustParse("1"),
 			corev1.ResourceMemory: resource.MustParse("1Gi"),
 		}
+
+		commands := []string{"/bin/sh", "-c"}
+		tmp := "/tmp/artifacts"
+		cache := "/cache/job.jar"
+		args := []string{
+			fmt.Sprintf("mkdir %s/lib", tmp),
+			fmt.Sprintf("gsutil cp %s %s/lib", strings.Join(urls[:], " "), tmp),
+			fmt.Sprintf("$(cd %s && zip -r job.jar .)", tmp),
+			fmt.Sprintf("cp /tmp/job.jar %s", cache),
+		}
+
 		//FIXME(regadas): this strategy will likely change
 		container := corev1.Container{
 			Name:      "gcs-downloader",
 			Image:     "google/cloud-sdk",
-			Command:   []string{"gsutil"},
-			Args:      []string{"cp", out.JarFile, "/cache/job.jar"},
+			Command:   commands,
+			Args:      args,
 			Resources: corev1.ResourceRequirements{Limits: resourceList},
 		}
-		out.JarFile = "/cache/job.jar"
+		out.JarFile = cache
 		out.InitContainers = append(out.InitContainers, container)
-		out.Volumes = append(out.Volumes, cacheVolume)
-		out.VolumeMounts = append(out.VolumeMounts, cacheVolumeMount)
```

Comment:
These volumes are needed in case the default config don't provide. I see what led to confusion. we should have this cache volume with probably a random suffix

Reply:
The tests fails with those bc. they are already added by the default config. This creates the same volume twice.

Reply:
right, the default config you are seeing is a test default config they might not exist in reality so we need to add volumes here. To make sure we don't conflict with others being set we should have a unique name for this volume.

Reply:
But this is can be addressed afterward this bug is not introduced by this change 😄

Reply:
Ok I'll revert that change then. The build will be broken thought :/